### PR TITLE
Stevesbranch

### PR DESCRIPTION
**Pull request by StevenBucher98**
Added the user interface, did not add the map fix yet but need to finish some other stuff (school shit) first, same goes with dependency injection.

Please tear this apart, suggestions and recommendations are much appreciated.

### CODE REVIEW

File: src/interaction.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+    close() {
+        this.rl.close();
+    }
+
+    // pass in readline
+    async confirmChanges(changeSet: any): Promise<boolean> {
+        const table = new Table({
+            head: ["Type", "Action", "Logistical Resource Id", "Resource Type"],
+            colWidths: [10, 20, 30, 30],
```
Comment (DonoA) [inline src/interaction.ts:47]: This should be set dynamically

**Review by DonoA — CHANGES_REQUESTED** [review]
Just one comment.

I do think it's interesting how many formatting things your editor changed. Are you using any special plugins with VS code?

Also if you don't have the editor config plugin that would be good to get because this project does have an editor config attached.

File: src/interaction.ts (hunk 1 above)
Comment (StevenBucher98) [reply src/interaction.ts:47]: I use Prettier as VScode plugin, do you mean that the table's columns should be dynamic?

Comment (DonoA) [reply src/interaction.ts:47]: ya, the table should fit the console window

**Pull request merged**